### PR TITLE
Update prometheus 3.0.1

### PR DESCRIPTION
**Pull request by Pokom**
:warning: This change is based off of #3 since I needed those changes to generate the new libsonnet files. ⚠️ 

- Updates go mod files to use Prometheus 3.0.1
- Regenerate the libsonnet to use Prometheus v3.0.1 release
- Updates the Makefile to provide the ability to define the Prometheus Version as an
environment variable.
- Updates configuration via `make schemas`
- Updates raw.libsonnet files via `make -B prometheusConfig/raw.libsonnet`

Right now there is a problem with this PR that I could use help with.
Updating the raw.libsonnet file results in the following error

```
RUNTIME ERROR: max stack frames exceeded.
        <std>:1545:5-33 function <anonymous>
        <std>:1524:15-38        object <anonymous>
        generator/vendor/github.com/crdsonnet/crdsonnet/crdsonnet/resolver.libsonnet:55:22-33thunk from <thunk from <function <anonymous>>>
        generator/vendor/github.com/crdsonnet/crdsonnet/crdsonnet/resolver.libsonnet:60:32-35thunk from <thunk <splitRef> from <function <anonymous>>>
        generator/vendor/github.com/crdsonnet/crdsonnet/crdsonnet/resolver.libsonnet:9:44-47 thunk from <thunk <fragmentSplit> from <function <anonymous>>>
        generator/vendor/github.com/crdsonnet/crdsonnet/crdsonnet/resolver.libsonnet:9:29-56 thunk <fragmentSplit> from <function <anonymous>>
        generator/vendor/github.com/crdsonnet/crdsonnet/crdsonnet/resolver.libsonnet:12:41-54thunk from <thunk <querySplit> from <function <anonymous>>>
        generator/vendor/github.com/crdsonnet/crdsonnet/crdsonnet/resolver.libsonnet:12:26-66thunk <querySplit> from <function <anonymous>>
        generator/vendor/github.com/crdsonnet/crdsonnet/crdsonnet/resolver.libsonnet:14:56-66thunk from <thunk from <thunk <hasScheme> from <function <anonymous>>>>
        <std>:1625:27-30        thunk from <function <anonymous>>
        ...
        <std>:32:25-26  thunk from <function <anonymous>>
        <std>:32:16-27  function <anonymous>
        generator/vendor/github.com/crdsonnet/astsonnet/main.libsonnet:818:14-32        object <anonymous>
        generator/vendor/github.com/crdsonnet/astsonnet/utils.libsonnet:47:9-26 thunk from <thunk <matches> from <function <anonymous>>>
        generator/vendor/github.com/crdsonnet/astsonnet/utils.libsonnet:(42:7)-(48:8)   thunk <matches> from <function <anonymous>>
        generator/vendor/github.com/crdsonnet/astsonnet/utils.libsonnet:50:19-26        thunk from <function <anonymous>>
        generator/vendor/github.com/crdsonnet/astsonnet/utils.libsonnet:50:8-27 function <anonymous>
        generator/main.libsonnet:(6:5)-(15:6)
        generator/main.libsonnet:17:3-6
        Top-level function call

<stdin>:1:1 Unexpected end of file
```

### CODE REVIEW

**Review by Duologic — APPROVED** [review]

**Review by Duologic — APPROVED** [review]

**Comment by Duologic** [comment]
bah, I though the GitHub UI 'resolve conflict' button would attempt rebase :-1:

**Comment by Duologic** [comment]
You'll also have to run the make targets for the raw.libsonnet files, Makefile dependencies have not been configured properly yet.

**Comment by Pokom** [comment]
@Duologic, mind checking the PR description for the output of running make for `raw.libsonnet` files? I get runtime error for max stack calls and I don't quite know how to debug that.

File: generator/config.json [hunk 1]
```diff
@@ -629,6 +725,9 @@
         },
         "azuread": {
           "$ref": "#/$defs/AzureAdconfig"
+        },
+        "google_iam": {
+          "$ref": "#/$defs/Config"
```
Comment (Duologic) [inline generator/config.json:737]: This is the problematic bit, the ref is resolves to the top-level config, causing endless recursion.

The generator for this schema doesn't handle duplicate naming of structs properly, causing this problem. It refers to this simple struct: https://github.com/prometheus/prometheus/blob/main/storage/remote/googleiam/googleiam.go#L28

We can easily account for this in the generator.

**Comment by Duologic** [comment]
I've pushed two commits, one that updates the Makefile and adds a fix for the commit, the other with all the generated files.

I'm using the Go version of jsonnet:
```
➜ jsonnet --version
Jsonnet commandline interpreter (Go implementation) v0.20.0
```

We could subsitute for jrsonnet to make it faster but it seems to run quite fast on my machine.

**Pull request merged**